### PR TITLE
[cammodels] fix and improve extractor

### LINKED ISSUE #22734: [cammodels] ExtractorError: Unable to find manifest URL root
- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2019.10.16**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log
LATEST VERSION (broken):
```
# ./youtube-dl -vg https://www.cammodels.com/cam/agnesss
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-vg', u'https://www.cammodels.com/cam/agnesss']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.10.16
[debug] Python version 2.7.16 (CPython) - Linux-5.3.6-050306-generic-x86_64-with-Ubuntu-19.04-disco
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, rtmpdump 2.4
[debug] Proxy map: {}
ERROR: Unable to find manifest URL root; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "./youtube-dl/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "./youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "./youtube-dl/youtube_dl/extractor/cammodels.py", line 43, in _real_extract
    raise ExtractorError(error, expected=expected)
ExtractorError: Unable to find manifest URL root; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```
LAST KNOWN WORKING:
```
# ./youtube-dl-lastok -vg https://www.cammodels.com/cam/agnesss
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-vg', u'https://www.cammodels.com/cam/agnesss']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.06.27
[debug] Python version 2.7.16 (CPython) - Linux-5.3.6-050306-generic-x86_64-with-Ubuntu-19.04-disco
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, rtmpdump 2.4
[debug] Proxy map: {}
[debug] Default format spec: bestvideo+bestaudio/best
rtmp://sea1c-edge-33.naiadsystems.com:1936/live/01ead87f-66cc-465c-9cca-5f91e2505708_2000_1280x720_56
```

## Description
* You need to use a model that is online: change the name in the URL accordingly
* One could make it work again by reverting the changes in utils.py ("random_user_agent"):
```
utils.py, line 1669:
   > 'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64; rv:59.0) Gecko/20100101 Firefox/59.0',
   < 'User-Agent': random_user_agent(),
 
```
* not sure if OS updates or web site changes made a difference because it has been working with a version of youtube-dl  _2019.06.27 < ok < 2019.09.28_ in September where utils.py was already the newer version
* no difference in using python2 or python3

Thanks for your work.

### PR DESCRIPTION
**Pull request by JChris246**
### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
This PR fixes the broken extractor as described in #22734 by adding the url to the manifest server as it no longer exists in the html. Additionally the jpeg formats are added as thumbnails.

Closes #22734

### CODE REVIEW

File: youtube_dl/extractor/cammodels.py [hunk 1]
```diff
@@ -84,6 +62,14 @@ def _real_extract(self, url):
...
                     })
+                elif format_id == 'jpeg':
+                    thumbnails.append({
+                        'url': f.get('url'),
+                        'width': f.get('width'),
+                        'height': f.get('height'),
+                        'format_id': f.get('format_id')
+                    })
+                    continue
                 else:
```
Comment (dirkf) [inline youtube_dl/extractor/cammodels.py:73]: Improve type checking, etc.
```suggestion
                else:
                    if format_id == 'jpeg':
                        t_url = url_or_none(f.get('url'))
                        if t_url:
                            thumbnails.append({
                                'url': t_url,
                                'width': int_or_none(f.get('width')),
                                'height': int_or_none(f.get('height')),
                                'format_id': f.get('format_id')
                            })
```

**Review by dirkf — CHANGES_REQUESTED** [review]
Thanks for your work!

I've made one suggestion.

I see there's no useful metadata in the non-JS webpage. I guess you're pulling all the available metadata of interest from the JSON?

File: youtube_dl/extractor/cammodels.py (hunk 1 above)
Comment (JChris246) [reply youtube_dl/extractor/cammodels.py:73]: Hey thanks for the review 👍.

I believe these values would have already been validated in lines 38 - 49. Implementing the suggested change would be redoing previously done checking.

Comment (dirkf) [reply youtube_dl/extractor/cammodels.py:73]: Oh, sure: but then you can say `f['url']`, etc.

**Review by dirkf — APPROVED** [review]

**Pull request merged**